ksql-rest-app/src/main/java/io/confluent/ksql/rest/util/ClusterTerminator.java

```diff
@@ -0,0 +1,112 @@
+/**
+ * Copyright 2018 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+package io.confluent.ksql.rest.util;
+
+import io.confluent.ksql.KsqlEngine;
+import io.confluent.ksql.metastore.KsqlTopic;
+import io.confluent.ksql.metastore.MetaStore;
+import io.confluent.ksql.rest.server.KsqlRestConfig;
+import io.confluent.ksql.util.ExecutorUtil;
+import io.confluent.ksql.util.KafkaTopicClient;
+import io.confluent.ksql.util.KsqlConfig;
+import io.confluent.ksql.util.KsqlException;
+import io.confluent.ksql.util.PersistentQueryMetadata;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+public class ClusterTerminator {
+
+  private final KsqlConfig ksqlConfig;
+  private final KsqlEngine ksqlEngine;
+
+  public ClusterTerminator(
+      final KsqlConfig ksqlConfig,
+      final KsqlEngine ksqlEngine
+  ) {
+    Objects.requireNonNull(ksqlConfig, "ksqlConfig is null.");
+    Objects.requireNonNull(ksqlEngine, "ksqlEngine is null.");
+    this.ksqlConfig = ksqlConfig;
+    this.ksqlEngine = ksqlEngine;
+  }
+
+  // Todo: Fail requesst if user requests non-sink topic to be deleted using non-regex
+  @SuppressWarnings("unchecked")
+  public void terminateCluster(final List<String> deleteTopicPatterns) {
+    ksqlEngine.stopAcceptingStatements();
+    terminateAllQueries();
+    deleteSinkTopics(deleteTopicPatterns);
+    deleteCommandTopic();
+  }
+
+  private void deleteSinkTopics(final List<String> deleteTopicPatterns) {
+    if (deleteTopicPatterns.isEmpty()) {
+      return;
+    }
+
+    final List<Pattern> patterns = deleteTopicPatterns.stream()
```

Comment:
Just a thought... but if the patterns included an invalid pattern, then an exception would be thrown at this point, but the `StatementExecutor` will have already have closed a lot of stuff....

So we should probably have the `/terminate` resource handler validate the passed in delete topic patterns are valid by doing a test compilation of them.